packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -238,7 +238,7 @@ function patchFragment(n1: VFragment, n2: VFragment, parent: ParentNode, rendere
     }
 
     // Note: not reusing n1.elm, because during patching, it may be patched with another text node.
-    n2.elm = n2.leading.elm;
+    n2.elm = n2.leading?.elm;
```

Comment:
Is there a way that the fragment can be empty?

If it could empty, then "leading" might be empty and there might be issues when computing the "anchor" (insert before/after) the fragment in update(Static/Dynamic)Children.